### ISSUE TITLE
cargo warning when updating grin crates

### ISSUE THREAD

**Issue by antiochp** [mimblewimble/grin-wallet#533]
Saw the following warning when updating grin crates to latest `master` on grin-wallet - 

```
cargo update -p grin_core
    Updating git repository `https://github.com/mimblewimble/grin`
    Updating crates.io index
warning: two git dependencies found for `https://github.com/mimblewimble/grin` 
where one uses `branch = "master"` and the other doesn't; 
this will break in a future version of Cargo, so please ensure the dependency forms are consistent
```

Have not investigated exactly what is going on. But we should fix this sooner rather than later.

**Comment by trevyn** [comment]
This appears to be fixed by 7896001 on Dec 1.

With `cargo` 1.50, I can reproduce the warning at the previous commit c3481a3, but not at 7896001.

The commit includes several copies of this change to `Cargo.lock`, adding `branch=master`:

```diff
- source = "git+https://github.com/mimblewimble/grin#48efb693e2e5cdf60449b7f009214491d5bf3646"
+ source = "git+https://github.com/mimblewimble/grin?branch=master#f48a23655ded65285ab412ad22b8f6a7a0075e56"
```

This is apparently related to a change in how `cargo` handles the default branch and the `master` vs `main` change, see https://github.com/rust-lang/cargo/pull/8522 for pointers to all of the exciting details.

**Comment by quentinlesceller** [comment]
Thanks for investigating @trevyn, I can also confirm it does not happen anymore.
